### ISSUE TITLE
Cypress test runner crashes when loading an image from another domain

### ISSUE THREAD

**Issue by megahertz** [cypress-io/cypress#5052]
### Current behavior:

- In a test1, page1 from domain1 loads an image from domain2
- in a test2, `cy.visit('https://domain2/page2')` crashes test runner when running all specs
- The bug occurs on HTTPS sites only, there is no such a problem on HTTP

### Steps to reproduce: (app code and test code)

https://github.com/megahertz/cy-bug

### Versions

cypress 3.4.1, linux or Docker, chromium 75 and electron 61

**Comment by megahertz** [comment]
The simpler way to reproduce:

```js
// test1.js
describe('Domain 1', () => {
  it('should load url', () => {
    cy.visit('https://mochajs.org/'); // Loads avatars from opencollective.com
  });
});
```

```js
// test2.js
describe('Domain 2', () => {
  it('should load url', () => {
    cy.visit('https://opencollective.com/');
  });
});
```

**Comment by jennifer-shehane** [comment]
The provided repo is not loading the urls being visited after following the local instructions. http://recordit.co/gX0bhQBbPs

<img width="550" alt="Screen Shot 2020-01-07 at 2 55 50 PM" src="https://user-images.githubusercontent.com/1271364/71880117-fd52c100-315d-11ea-88b6-55b339f4b4df.png">

The other example is sometimes failing to visit the `opencollective` url. Is this the behavior you are describing? I cannot get Cypress to 'crash' as you've described.

![](http://g.recordit.co/gX0bhQBbPs.gif)

**Comment by megahertz** [comment]
@jennifer-shehane Thank you. Yes, it's similar to #3975. There is only one difference. In #3975, 404 error is displayed only inside a frame. But in my situation, the whole page of Cypress runner is redirected to `/__/`, so Cypress doesn't work until it's restarted.

As for ENOTFOUND, I forgot to mention that test domains should be added to host file explicitly in most environments:
```sh
echo '127.0.0.1 domain1.localhost' >> /etc/hosts
echo '127.0.0.1 domain2.localhost' >> /etc/hosts
```

So it's easier to run the test script inside Docker container where the domains are already defined.

**Comment by jennifer-shehane** [comment]
Duplicate of #3975

**Comment by jennifer-shehane** [comment]
This fix is available starting in [`4.6.0`](https://docs-staging.cypress.io/guides/references/changelog.html#4-6-0) as an [experiment](https://on.cypress.io/experiments) which you can access by setting this config option in your `cypress.json` or elsewhere:

```json
{
	"experimentalSourceRewriting": true
}
```

The fix is experimental, so there may be some situations where the this is not fixed. 

If you're still this issue while setting the `experimentalSourceRewriting` to true in 4.6.0 - [open a new issue](https://github.com/cypress-io/cypress/issues/new) with a reproducible example + screenshots, etc - filling out our issue template.
